src/Compilers/Core/Portable/MetadataReader/MetadataDecoder.cs

```diff
@@ -1174,21 +1182,27 @@ internal ImmutableArray<LocalInfo<TypeSymbol>> GetLocalInfo(StandaloneSignatureH
         }
 
         /// <exception cref="UnsupportedSignatureContent">If the encoded parameter type is invalid.</exception>
-        private void DecodeParameterOrThrow(ref BlobReader signatureReader, /*out*/ ref ParamInfo<TypeSymbol> info)
+        private void DecodeParameterOrThrow(ref BlobReader signatureReader, /*out*/ ref ParamInfo<TypeSymbol> info, bool isReturn)
         {
+            var allowedRequiredModifiers = AllowedRequiredModifierType.System_Runtime_InteropServices_InAttribute;
+            if (isReturn)
+            {
+                allowedRequiredModifiers |= AllowedRequiredModifierType.System_Runtime_CompilerServices_IsExternalInit;
```

Comment:
>allowedRequiredModifiers |= AllowedRequiredModifierType.System_Runtime_CompilerServices_IsExternalInit [](start = 16, length = 102)

According to the speclet this modifier is allowed only on a property setter. It doesn't look like this requirement is enforced anywhere.

Reply:
Indeed, thanks. Marked with PROTOTYPE. I don't have a solution to tighten this.

---
In reply to: [413385302](https://github.com/dotnet/roslyn/pull/43275#discussion_r413385302) [](ancestors = 413385302)